cmd/internal/initializer/initializer.go

```diff
@@ -197,11 +212,21 @@ func (i *Initializer) Restore(ctx context.Context, version *providers.BackupVers
 		return fmt.Errorf("could not delete priorly downloaded file: %w", err)
 	}
 
-	err := i.bp.DownloadBackup(ctx, version)
+	backupFilePath, err := i.bp.DownloadBackup(ctx, version, constants.DownloadDir)
 	if err != nil {
 		return fmt.Errorf("unable to download backup: %w", err)
 	}
 
+	if i.encrypter != nil {
+		if encryption.IsEncrypted(backupFilePath) {
+			backupFilePath, err = i.encrypter.Decrypt(backupFilePath)
+			if err != nil {
+				return fmt.Errorf("unable to decrypt backup: %w", err)
+			}
+		}
+		i.log.Info("restoring unencrypted backup with configured encryption - skipping decryption...")
```

Comment:
This log line seems to be wrong because it's always shown. No matter if it was encrypted or not.